src/wallet/cbf.rs

```diff
@@ -0,0 +1,88 @@
+use std::{collections::HashMap, net::SocketAddr, path::PathBuf, thread, time::Duration};
+use std::cell::Cell;
+
+use nakamoto::{
+    client::{
+        chan::Receiver,
+        Client,
+        Config,
+        Event,
+        Handle as ClientHandle,
+        handle::Handle,
+    },
+    net::poll::Waker,
+};
+use nakamoto::chain::Transaction;
+use nakamoto::p2p::fsm::fees::FeeEstimate;
+
+type Reactor = nakamoto::net::poll::Reactor<std::net::TcpStream>;
+
+pub struct CbfBlockchain {
+    receiver: Receiver<Event>,
+    client_handle: ClientHandle<Waker>,
+    timeout: Duration,
+    fee_data: Cell<HashMap<u32, FeeEstimate>>,
+    broadcasted_txs: Cell<Vec<Transaction>>,
+    last_sync_height: Cell<u32>,
+}
+
+pub enum CbfSyncError {
+    FilterHeaderRetrievalError,
+    BlockFilterDownloadError,
+    TransactionRetrievalError,
```

Comment:
All of these errors should be coming from the Nakamoto client error. So instead of making different enum variants for them, just add one `NakamotoError(nakamoto::client::error)`.

Reply:
Acknowledged in 055c0c44829bd322b79205ec38655b7207b6d3e8